activity/src/main/java/com/amazonaws/stepfunctions/cloudformation/activity/DeleteHandler.java

```diff
@@ -1,49 +1,136 @@
 
-            proxy.injectCredentialsAndInvoke(deleteActivityRequest, sfnClient::deleteActivity);
+        DeleteActivityRequest deleteActivityRequest = new DeleteActivityRequest();
+        deleteActivityRequest.setActivityArn(model.getArn());
 
-            return ProgressEvent.<ResourceModel, CallbackContext>builder()
-                    .status(OperationStatus.SUCCESS)
-                    .build();
-        } catch (Exception e) {
-            logger.log("ERROR Deleting Activity, caused by " + e.toString());
-            return handleDefaultError(request, e);
+        proxy.injectCredentialsAndInvoke(deleteActivityRequest, sfnClient::deleteActivity);
+
+        return stabilizeDeleteActivity(proxy, model, 0, logger);
+    }
+
+    private ProgressEvent<ResourceModel, CallbackContext> stabilizeDeleteActivity(
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceModel model,
+            final int currentRetryCount,
+            final Logger logger) {
+
+        ClientConfiguration clientConfiguration = new ClientConfiguration()
+                .withRetryPolicy(PredefinedRetryPolicies.NO_RETRY_POLICY);
+
+        DescribeActivityRequest describeActivityRequest = new DescribeActivityRequest();
+        describeActivityRequest.setActivityArn(model.getArn());
+
+        for (int i = 0; i < STABILIZATION_CALL_COUNT; i++) {
+            try {
+                AWSStepFunctions sfnClient = AWSStepFunctionsClientBuilder.standard()
+                        .withClientConfiguration(clientConfiguration)
+                        .build();
+
+                proxy.injectCredentialsAndInvoke(describeActivityRequest, sfnClient::describeActivity);
+                // If DescribeActivity succeeds, the activity still exists.
+                logger.log("Activity still exists after DescribeActivity attempt #" + (i + 1));
+                // If any DescribeActivity call succeeded, the activity still exists and the cache has not been cleared yet
+                // Retry as DescribeActivity failure count has not reached the desired stabilization attempts
+                return ProgressEvent.<ResourceModel, CallbackContext>builder()
+                        .status(OperationStatus.IN_PROGRESS)
+                        .callbackContext(CallbackContext.builder()
+                                .retryCount(currentRetryCount)
+                                .isActivityDeletionStarted(true)
+                                .build())
+                        .callbackDelaySeconds(calculateExponentialBackoffDelayWithJitter(currentRetryCount))
+                        .build();
+            } catch (Exception e) {
+                // If DescribeActivity fails, wait for the next attempt.
+                logger.log("DescribeActivity failed on attempt #" + (i + 1) + ": " + e.getMessage());
+                Uninterruptibles.sleepUninterruptibly(10, TimeUnit.MILLISECONDS);
+            }
```

Comment:
If we do 3 retries, waiting up to a maximum of 30s on the final attempt, we will exceed the 60s timeout within which we have to return a progress event. 

Could we implement this such that the CallbackContext maintains a count of the number of successive DescribeActivity attempts that return with the activity not existing? If we get a DescribeActivity call that returns as a `200`, then we reset the count to `0`.